src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.cpp

```diff
@@ -31,27 +32,235 @@
+
+    const QMap<char, QByteArray> alphabetChars = createDnaAlphabetResolutionMap();
+
+    foreach (const char i, alphabetChars.keys()) {
+        foreach (const char j, alphabetChars.keys()) {
+            if (0 == res[i][j]) {
+                // Unambiguous nucleotide pairs are already registered
+                // If at least one nucleotide in pair is ambiguous, then the pair value should be an avarage value of all possible variants.
+                int value = 0;
+                for (int k = 0; k < alphabetChars[i].length(); ++k) {
+                    for (int m = 0; m < alphabetChars[j].length(); ++m) {
+                        char char1 = alphabetChars[i][k];
+                        char char2 = alphabetChars[j][m];
+                        value += res[char1][char2];
+                    }
+                }
+                const int count = alphabetChars[i].length() * alphabetChars[j].length();
+                value /= count;
+                res[i][j] = value;
+            }
+        }
+    }
+
+    return res;
+}
+
+static MononucleotidesExtinctionCoefficientsMap createRnaMononucleotidesExtinctionCoefficients() {
+    MononucleotidesExtinctionCoefficientsMap res(MAP_SIZE, 0);
+    res['A'] = 15400;
+    res['C'] = 7200;
+    res['G'] = 11500;
+    res['U'] = 9900;
+    return res;
+}
+
+static DinucleotidesExtinctionCoefficientsMap createRnaDinucleotidesExtinctionCoefficients() {
+    DinucleotidesExtinctionCoefficientsMap res(MAP_SIZE, QVector<int>(MAP_SIZE, 0));
+
+    res['A']['A'] = 27400;
+    res['A']['C'] = 21000;
+    res['A']['G'] = 25000;
+    res['A']['U'] = 24000;
+
+    res['C']['A'] = 21000;
+    res['C']['C'] = 14200;
+    res['C']['G'] = 17800;
+    res['C']['U'] = 16200;
+
+    res['G']['A'] = 25200;
+    res['G']['C'] = 17400;
+    res['G']['G'] = 21600;
+    res['G']['U'] = 21200;
+
+    res['U']['A'] = 24600;
+    res['U']['C'] = 17200;
+    res['U']['G'] = 20000;
+    res['U']['U'] = 19600;
+
+    const QMap<char, QByteArray> alphabetChars = createRnaAlphabetResolutionMap();
+
+    foreach (const char i, alphabetChars.keys()) {
+        foreach (const char j, alphabetChars.keys()) {
+            if (0 == res[i][j]) {
+                // Unambiguous nucleotide pairs are already registered
+                // If at least one nucleotide in pair is ambiguous, then the pair value should be an avarage value of all possible variants.
+                int value = 0;
+                for (int k = 0; k < alphabetChars[i].length(); ++k) {
+                    for (int m = 0; m < alphabetChars[j].length(); ++m) {
+                        char char1 = alphabetChars[i][k];
+                        char char2 = alphabetChars[j][m];
+                        value += res[char1][char2];
+                    }
+                }
+                const int count = alphabetChars[i].length() * alphabetChars[j].length();
+                value /= count;
+                res[i][j] = value;
+            }
+        }
+    }
```

Comment:
These lines copy-pastes the same code from the createDnaDinucleotidesExtinctionCoefficients function. I think, it's better to move it to the separate function